lib/common/bits.h

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ * You may select, at your option, one of the above-listed licenses.
+ */
+
+#ifndef ZSTD_BITS_H
+#define ZSTD_BITS_H
+
+#include "mem.h"
+
+MEM_STATIC unsigned ZSTD_highbit32_fallback(U32 val) {
+    static const U32 DeBruijnClz[32] = {  0,  9,  1, 10, 13, 21,  2, 29,
+                                         11, 14, 16, 18, 22, 25,  3, 30,
+                                          8, 12, 20, 28, 15, 17, 24,  7,
+                                         19, 27, 23,  6, 26,  5,  4, 31 };
+    val |= val >> 1;
+    val |= val >> 2;
+    val |= val >> 4;
+    val |= val >> 8;
+    val |= val >> 16;
+    return DeBruijnClz[(val * 0x07C4ACDDU) >> 27];
+}
+
+MEM_STATIC unsigned ZSTD_highbit32(U32 val)   /* compress, dictBuilder, decodeCorpus */
+{
+    assert(val != 0);
+    {
+#   if defined(_MSC_VER)   /* Visual */
+#       if STATIC_BMI2 == 1
+            return _lzcnt_u32(val)^31;
+#       else
+            if (val != 0) {
+                unsigned long r;
+                _BitScanReverse(&r, val);
+                return (unsigned)r;
+            } else {
+                /* Should not reach this code path */
+                __assume(0);
+            }
```

Comment:
This can be in this way, https://godbolt.org/ shows the same assembly code generated. 
```c
        unsigned long r;
        _BitScanReverse(&r, val);
        return (U32) r;
```
Other MSVC code can also be simplified like this.